File: 3351-maximize-happiness-of-selected-children/maximize-happiness-of-selected-children.js
/**
 * @param {number[]} happiness
 * @param {number} k
 * @return {number}
 */

// brute force 
var maximumHappinessSum = function(happiness, k) {
    happiness = happiness.sort((a,b)=>b-a);
    let ans = 0;
    for(let i=0; i<k; i++){
        ans += Math.max(happiness[i]-i,0)    
    }
    return ans;
};

File: 3351-maximize-happiness-of-selected-children/maximize-happiness-of-selected-children.rs
impl Solution {
    pub fn maximum_happiness_sum(happiness: Vec<i32>, k: i32) -> i64 {
        let mut happiness = happiness;
        happiness.sort_unstable_by(|a, b| b.cmp(a));
        let mut ans = 0i64;
        for i in 0..k as usize {
            ans += std::cmp::max(happiness[i] - i as i32, 0) as i64;
        }
        ans
    }
}